kool-core/src/jvmMain/kotlin/de/fabmax/kool/platform/gl/LoadedTextureGl.kt

```diff
@@ -65,16 +66,38 @@ class LoadedTextureGl(val ctx: Lwjgl3Context, val target: Int, val texture: Int,
             // swap byte order (rgba -> abgr)
             val bi = i * 4
             val r = pixels[bi].toUByte().toInt()
-            val g = pixels[bi+1].toUByte().toInt()
-            val b = pixels[bi+2].toUByte().toInt()
-            val a = pixels[bi+3].toUByte().toInt()
+            val g = pixels[bi + 1].toUByte().toInt()
+            val b = pixels[bi + 2].toUByte().toInt()
+            val a = pixels[bi + 3].toUByte().toInt()
             val rgba = (a shl 24) or (r shl 16) or (g shl 8) or b
             // todo: setting individual pixels is rather slow
             img.setRGB(i % width, height - 1 - i / width, rgba)
         }
         return img
     }
 
+    override fun readTexturePixels(targetData: TextureData) {
+        var error: Int?
+        while( run { error = glGetError(); error } != GL_NO_ERROR ) {
+            logW { "GL Error before run: ${error?.toString(16)}" }
+        }
+        glActiveTexture(texture)
```

Comment:
[glActiveTexture()](https://registry.khronos.org/OpenGL-Refpages/gl4/html/glActiveTexture.xhtml) expects a texture unit enum (something like `GL_TEXTURE0`) instead of the texture name / id. It should be ok to use some fixed value here or just delete the line entirely.